src/main/java/graphql/schema/TypeResolver.java

```diff
@@ -1,9 +1,15 @@
 package graphql.schema;
 
 
+import graphql.TypeResolutionEnvironment;
```

Comment:
Can we separate out this change regarding extension of the TypeResolver interface? That conflates the original goal: resolving TypeReferences.... I'm also curious about why you need all this extra context information when you implement your TypeResolver.